src/agentic/tools/meeting_tool.py

```diff
@@ -0,0 +1,493 @@
+                )  
+                
+                if response.status_code == 200:  
+                    data = response.json()  
+                    transcripts = json.dumps(data.get("bot_data", {}).get("transcripts", []))  
+                    meeting.transcript = transcripts  
+                    session.commit()  
+                else:  
+                    return {"status": "error", "message": f"Failed to fetch transcript: {response.text}"}  
+            
+            return {"status": "success", "transcript": json.loads(meeting.transcript)}  
+        
+        except Exception as e:  
+            return {"status": "error", "message": f"Error fetching transcript: {str(e)}"}  
+
+
+    def get_summary(self, meeting_id: str) -> dict:
+        """Generate a detailed summary of the meeting"""
+        try:
+            transcript_result = self.get_transcript(meeting_id)
+            if transcript_result["status"] == "error":
+                return transcript_result
+                
+            transcript = transcript_result["transcript"]
+            
+            formatted_transcript = ""
+            for entry in transcript:
+                if entry.get('words'):
+                    words = ' '.join([word['text'] for word in entry['words']])
+                    speaker = entry['speaker']
+                    formatted_transcript += f"{speaker}: {words}\n"
+
+            system_prompt = """
+                                Please provide a detailed summary of this meeting transcript. Include:
+                                1. Main topics discussed
+                                2. Key decisions made
+                                3. Action items and assignments
+                                4. Important points raised by each participant
+                                5. Timeline of major discussion points
+                                Format the summary in clear sections with headers.
+
+                                """
+
+            client = OpenAI(api_key=self.openai_api_key)
+
+            response = client.beta.chat.completions.parse(
+                model="gpt-4o",
+                messages=[
+                    {"role": "system", "content": system_prompt},
+                    {"role": "user", "content": formatted_transcript},
+                ]
+            )
+            
+            detailed_summary = response.choices[0].message.parsed
+            
+            session = self._get_session()
+            meeting = session.query(Meeting).filter_by(id=meeting_id).first()
+            if meeting:
+                meeting.summary = detailed_summary
+                session.commit()
+            
+            # Initialize RAG if needed
+            self._initialize_rag()
+            
+            # Use chonkie for semantic chunking (imported via rag_helper)
+            from chonkie import SemanticChunker
+            chunker = SemanticChunker(threshold=0.5, delim=[". ", "! ", "? ", "\n"])
```

Comment:
Could you not just use this helper method: https://github.com/supercog-ai/agentic/blob/main/src/agentic/utils/rag_helper.py#L147 ?

It will be nice if we can centralize "basic RAG" operations so that other people building tools can use RAG without knowing about chonie, weaviate, etc...